locobot/label_propogation/label_propogation_parallel.py

```diff
@@ -0,0 +1,341 @@
+# used for visualizing how well the label propogation will work
+# currenlty will use robot trajectory to realize
+# will rely on habitat dat to work on it
+
+# steps we will perform for label propogation
+# 1. Get point cloud
+# 2. Transpose the point cloud based on robot location
+# 3. Project the point cloud back the images
+from tokenize import String
+import numpy as np
+import os
+import cv2
+import json
+from copy import deepcopy as copy
+from IPython import embed
+import sys
+import time
+import ray
+from scipy.spatial.transform import Rotation
+from pycocotools.coco import COCO
+import glob
+import argparse
+
+
+# this function is implemented at 'from locobot.agent.locobot_mover_utils import transform_pose'
+# however ray was having toruble finding the function
+def transform_pose(XYZ, current_pose):
+    """
+    Transforms the point cloud into geocentric frame to account for
+    camera position
+    Input:
+        XYZ                     : ...x3
+    current_pose            : camera position (x, y, theta (radians))
+    Output:
+        XYZ : ...x3
+    """
+    R = Rotation.from_euler("Z", current_pose[2]).as_matrix()
+    XYZ = np.matmul(XYZ.reshape(-1, 3), R.T).reshape((-1, 3))
+    XYZ[:, 0] = XYZ[:, 0] + current_pose[0]
+    XYZ[:, 1] = XYZ[:, 1] + current_pose[1]
+    return XYZ
+
+
+@ray.remote
+def propogate_label(
+    root_path: str,
+    src_img_indx: int,
+    src_label: np.ndarray,
+    propogation_step: int,
+    base_pose_data: np.ndarray,
+    out_dir: str,
+):
+    """Take the label for src_img_indx and propogate it to [src_img_indx - propogation_step, src_img_indx + propogation_step]
```

Comment:
add more details here - a textual explanation of the approach basically, something like a pseudo code. Does this need depth for the entire trajectory? Also, what is the idea behind the depth filtering?